crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1039,6 +1039,126 @@ impl FromReflect for Cow<'static, str> {
     }
 }
 
+impl<T: FromReflect + Clone + Send + Sync> Array for Cow<'static, [T]> {
+    fn get(&self, index: usize) -> Option<&dyn Reflect> {
+        self.as_ref().get(index).map(|x| x as &dyn Reflect)
+    }
+
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect> {
+        self.to_mut().get_mut(index).map(|x| x as &mut dyn Reflect)
+    }
+
+    fn len(&self) -> usize {
+        self.as_ref().len()
+    }
+
+    fn iter(&self) -> ArrayIter {
+        ArrayIter::new(self)
+    }
+
+    fn drain(self: Box<Self>) -> Vec<Box<dyn Reflect>> {
+        self.iter().map(Reflect::clone_value).collect()
```

Comment:
You can save it from doing an unnecessary clone if you have a `Cow::Owned` by switching this to `self.into_owned().into_iter()`. That would only clone the data if it's currently borrowed, otherwise it would reuse the owned data. Although that currently triggers a false-positive in clippy, so it will need an `#[allow]` tag added.

Reply:
Thanks for pointing that out! Also, thank you for the allow tag idea. I was having lots of trouble getting around that Clippy lint.